source/main.cpp

```diff
@@ -114,16 +109,18 @@ int main(int argc, char *argv[])
 			testToRunName = *it;
 		else if(arg == "--tests")
 			printTests = true;
-		else if(arg == "-s" || arg == "--ships")
-			printShips = true;
-		else if(arg == "-w" || arg == "--weapons")
-			printWeapons = true;
+		else if(arg == "-s" || arg == "--ships" || arg == "-w" || arg == "--weapons" ||
```

Comment:
Maybe we should ask the printdata class if its concerned by this argument? Like with a method there? Caus this seems like a huge if in the main file.
What I mean is something like 'else if(PrintData::IsPrintArgument(arg)'

An alternative is just removing this, remove the printData variable, and always call PrintData::Print(argv) that would not do anything if there is no argument is supports (could return a bool or smth even though we wouldnt do much with it?)

Reply:
Passing the class the arguments to check if it's interested sounds like a good idea. I'll do that.
We can't just call PrintData::Print() here and expect it to run on its own because the data is loaded later on in main.

Reply:
please do this thanks